cordformation/src/main/kotlin/net/corda/plugins/Baseform.kt

```diff
@@ -172,4 +174,36 @@ open class Baseform : DefaultTask() {
             return false
         }
     }
+
+    /**
+     * This classloader ensures that classes in the Gradle project cannot access anything on
+     * the Gradle plugins classpath. The only exceptions to this are the net.corda.cordform.*
+     * classes and their dependencies (Kotlin and Typesafe Config), because these classes
+     * exist on both the project and the plugins classpaths.
+     *
+     * This plugin MUST load every net.corda.cordform.* class from the same classloader to
+     * ensure consistency. E.g. two [CordformDefinition] classes are assignment-incompatible
+     * if they have different classloaders.
+     *
+     * Note that this classloader's parent is the system classloader.
+     */
+    class CordformClassLoader(
+        urls: Array<URL>,
+        private val logger: Logger,
+        cordform: Class<*> = CordformDefinition::class.java // parameterised for testing
+)   : URLClassLoader(urls) {
+        private val pluginClassloader: ClassLoader = cordform.classLoader
+        private val prefixes: List<String>
+                      = listOf(cordform.`package`.name + '.', "com.typesafe.config.", "org.jetbrains.kotlin.")
```

Comment:
Please also add a comment to the cordform project's build.gradle to tell people about this line needing to be updated if new packages can be added.